## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 514647`
Commit: [`b6957ac0d8633ac461821e8cdd3605995e0897fc`](https://github.com/NixOS/nixpkgs/commit/b6957ac0d8633ac461821e8cdd3605995e0897fc) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/b6957ac0d8633ac461821e8cdd3605995e0897fc..pull/514647/head))
Merge: [`911fdc188770858c112316155c9d640a5e7f3382`](https://github.com/NixOS/nixpkgs/commit/911fdc188770858c112316155c9d640a5e7f3382)

Logs: https://github.com/kilyanni/nixpkgs-review-gha/actions/runs/25185005781


---
### `x86_64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python313Packages.cyscale</li>
    <li>python313Packages.cyscale.dist</li>
    <li>python314Packages.cyscale</li>
    <li>python314Packages.cyscale.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python313Packages.cyscale</summary>
<pre>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
test/test_type_registry.py:21: in &lt;module&gt;
    from scalecodec.types import Extrinsic
scalecodec/__init__.py:2: in &lt;module&gt;
    from .types import *
E   ModuleNotFoundError: No module named &#x27;scalecodec.types&#x27;
=========================== short test summary info ============================
ERROR test/test_extrinsic_payload.py
ERROR test/test_legacy_events.py
ERROR test/test_metadata.py
ERROR test/test_runtime_call.py
ERROR test/test_runtime_configuration.py
ERROR test/test_scale_info.py
ERROR test/test_scale_types.py
ERROR test/test_scalebytes.py
ERROR test/test_ss58.py
ERROR test/test_type_encoding.py
ERROR test/test_type_registry.py
!!!!!!!!!!!!!!!!!!! Interrupted: 11 errors during collection !!!!!!!!!!!!!!!!!!!
============================== 11 errors in 0.27s ==============================</pre>
</details>
<details>
<summary>python314Packages.cyscale</summary>
<pre>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
test/test_type_registry.py:21: in &lt;module&gt;
    from scalecodec.types import Extrinsic
scalecodec/__init__.py:2: in &lt;module&gt;
    from .types import *
E   ModuleNotFoundError: No module named &#x27;scalecodec.types&#x27;
=========================== short test summary info ============================
ERROR test/test_extrinsic_payload.py
ERROR test/test_legacy_events.py
ERROR test/test_metadata.py
ERROR test/test_runtime_call.py
ERROR test/test_runtime_configuration.py
ERROR test/test_scale_info.py
ERROR test/test_scale_types.py
ERROR test/test_scalebytes.py
ERROR test/test_ss58.py
ERROR test/test_type_encoding.py
ERROR test/test_type_registry.py
!!!!!!!!!!!!!!!!!!! Interrupted: 11 errors during collection !!!!!!!!!!!!!!!!!!!
============================== 11 errors in 0.33s ==============================</pre>
</details>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python313Packages.cyscale</li>
    <li>python313Packages.cyscale.dist</li>
    <li>python314Packages.cyscale</li>
    <li>python314Packages.cyscale.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>python313Packages.cyscale</summary>
<pre>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
test/test_type_registry.py:21: in &lt;module&gt;
    from scalecodec.types import Extrinsic
scalecodec/__init__.py:2: in &lt;module&gt;
    from .types import *
E   ModuleNotFoundError: No module named &#x27;scalecodec.types&#x27;
=========================== short test summary info ============================
ERROR test/test_extrinsic_payload.py
ERROR test/test_legacy_events.py
ERROR test/test_metadata.py
ERROR test/test_runtime_call.py
ERROR test/test_runtime_configuration.py
ERROR test/test_scale_info.py
ERROR test/test_scale_types.py
ERROR test/test_scalebytes.py
ERROR test/test_ss58.py
ERROR test/test_type_encoding.py
ERROR test/test_type_registry.py
!!!!!!!!!!!!!!!!!!! Interrupted: 11 errors during collection !!!!!!!!!!!!!!!!!!!
============================== 11 errors in 0.26s ==============================</pre>
</details>
<details>
<summary>python314Packages.cyscale</summary>
<pre>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
test/test_type_registry.py:21: in &lt;module&gt;
    from scalecodec.types import Extrinsic
scalecodec/__init__.py:2: in &lt;module&gt;
    from .types import *
E   ModuleNotFoundError: No module named &#x27;scalecodec.types&#x27;
=========================== short test summary info ============================
ERROR test/test_extrinsic_payload.py
ERROR test/test_legacy_events.py
ERROR test/test_metadata.py
ERROR test/test_runtime_call.py
ERROR test/test_runtime_configuration.py
ERROR test/test_scale_info.py
ERROR test/test_scale_types.py
ERROR test/test_scalebytes.py
ERROR test/test_ss58.py
ERROR test/test_type_encoding.py
ERROR test/test_type_registry.py
!!!!!!!!!!!!!!!!!!! Interrupted: 11 errors during collection !!!!!!!!!!!!!!!!!!!
============================== 11 errors in 0.29s ==============================</pre>
</details>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python313Packages.cyscale</li>
    <li>python313Packages.cyscale.dist</li>
    <li>python314Packages.cyscale</li>
    <li>python314Packages.cyscale.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>python313Packages.cyscale</summary>
<pre>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
test/test_type_registry.py:21: in &lt;module&gt;
    from scalecodec.types import Extrinsic
scalecodec/__init__.py:2: in &lt;module&gt;
    from .types import *
E   ModuleNotFoundError: No module named &#x27;scalecodec.types&#x27;
=========================== short test summary info ============================
ERROR test/test_extrinsic_payload.py
ERROR test/test_legacy_events.py
ERROR test/test_metadata.py
ERROR test/test_runtime_call.py
ERROR test/test_runtime_configuration.py
ERROR test/test_scale_info.py
ERROR test/test_scale_types.py
ERROR test/test_scalebytes.py
ERROR test/test_ss58.py
ERROR test/test_type_encoding.py
ERROR test/test_type_registry.py
!!!!!!!!!!!!!!!!!!! Interrupted: 11 errors during collection !!!!!!!!!!!!!!!!!!!
============================== 11 errors in 0.79s ==============================</pre>
</details>
<details>
<summary>python314Packages.cyscale</summary>
<pre>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
test/test_type_registry.py:21: in &lt;module&gt;
    from scalecodec.types import Extrinsic
scalecodec/__init__.py:2: in &lt;module&gt;
    from .types import *
E   ModuleNotFoundError: No module named &#x27;scalecodec.types&#x27;
=========================== short test summary info ============================
ERROR test/test_extrinsic_payload.py
ERROR test/test_legacy_events.py
ERROR test/test_metadata.py
ERROR test/test_runtime_call.py
ERROR test/test_runtime_configuration.py
ERROR test/test_scale_info.py
ERROR test/test_scale_types.py
ERROR test/test_scalebytes.py
ERROR test/test_ss58.py
ERROR test/test_type_encoding.py
ERROR test/test_type_registry.py
!!!!!!!!!!!!!!!!!!! Interrupted: 11 errors during collection !!!!!!!!!!!!!!!!!!!
============================== 11 errors in 0.80s ==============================</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python313Packages.cyscale</li>
    <li>python313Packages.cyscale.dist</li>
    <li>python314Packages.cyscale</li>
    <li>python314Packages.cyscale.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python313Packages.cyscale</summary>
<pre>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
test/test_type_registry.py:21: in &lt;module&gt;
    from scalecodec.types import Extrinsic
scalecodec/__init__.py:2: in &lt;module&gt;
    from .types import *
E   ModuleNotFoundError: No module named &#x27;scalecodec.types&#x27;
=========================== short test summary info ============================
ERROR test/test_extrinsic_payload.py
ERROR test/test_legacy_events.py
ERROR test/test_metadata.py
ERROR test/test_runtime_call.py
ERROR test/test_runtime_configuration.py
ERROR test/test_scale_info.py
ERROR test/test_scale_types.py
ERROR test/test_scalebytes.py
ERROR test/test_ss58.py
ERROR test/test_type_encoding.py
ERROR test/test_type_registry.py
!!!!!!!!!!!!!!!!!!! Interrupted: 11 errors during collection !!!!!!!!!!!!!!!!!!!
============================== 11 errors in 0.26s ==============================</pre>
</details>
<details>
<summary>python314Packages.cyscale</summary>
<pre>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
test/test_type_registry.py:21: in &lt;module&gt;
    from scalecodec.types import Extrinsic
scalecodec/__init__.py:2: in &lt;module&gt;
    from .types import *
E   ModuleNotFoundError: No module named &#x27;scalecodec.types&#x27;
=========================== short test summary info ============================
ERROR test/test_extrinsic_payload.py
ERROR test/test_legacy_events.py
ERROR test/test_metadata.py
ERROR test/test_runtime_call.py
ERROR test/test_runtime_configuration.py
ERROR test/test_scale_info.py
ERROR test/test_scale_types.py
ERROR test/test_scalebytes.py
ERROR test/test_ss58.py
ERROR test/test_type_encoding.py
ERROR test/test_type_registry.py
!!!!!!!!!!!!!!!!!!! Interrupted: 11 errors during collection !!!!!!!!!!!!!!!!!!!
============================== 11 errors in 0.34s ==============================</pre>
</details>
</details>

--- a/pkgs/development/python-modules/cyscale/default.nix
+++ b/pkgs/development/python-modules/cyscale/default.nix
@@ -1,0 +1,53 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  more-itertools,
+  base58,
+  pytestCheckHook,
+}:
+
+buildPythonPackage (finalAttrs: {
+  pname = "cyscale";
+  version = "0.3.2";
+  pyproject = true;
+
+  __structuredAttrs = true;
+
+  src = fetchFromGitHub {
+    owner = "latent-to";
+    repo = "cyscale";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-L75xuo4LWfTMs1XYV8zSPtmxqgjWin9wcALDUjz5L1k=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    more-itertools
+    base58
+  ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  # test_valid_type_registry_presets walks ../scalecodec/type_registry relative
+  # to the test file, which no longer exists after removing the source package dir
+  disabledTests = [ "test_valid_type_registry_presets" ];
+
+  # remove source package so tests import the installed Cython extensions, not the .pyx sources
+  preCheck = ''
+    rm -rf scalecodec
+  '';
+
+  pythonImportsCheck = [ "scalecodec" ];
+
+  meta = {
+    description = "Cython-accelerated SCALE codec library for Substrate-based blockchains (Polkadot, Kusama, Bittensor, etc.)";
+    longDescription = "A drop-in replacement for py-scale-codec — same scalecodec module name, same public API, compiled with Cython for improved throughput.";
+    homepage = "https://github.com/latent-to/cyscale";
+    changelog = "https://github.com/latent-to/cyscale/releases/tag/${finalAttrs.src.tag}";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ kilyanni ];
+  };
+})

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -3490,6 +3490,8 @@ self: super: with self; {
 
   cyrtranslit = callPackage ../development/python-modules/cyrtranslit { };
 
+  cyscale = callPackage ../development/python-modules/cyscale { };
+
   cysignals = callPackage ../development/python-modules/cysignals { };
 
   cython = callPackage ../development/python-modules/cython { };

